plugins/modules/aci_fabric_external_routing_profile.py

```diff
@@ -0,0 +1,308 @@
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        name=dict(type="str", aliases=["routing_profile", "profile"]),
+        fabric_id=dict(type="int", aliases=["fabric", "fid"]),
+        description=dict(type="str", aliases=["descr"]),
+        subnets=dict(type="list", elements="str", aliases=["ip_addresses", "ips"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["fabric_id", "name"]],
+            ["state", "present", ["fabric_id", "name"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    fabric_id = module.params.get("fabric_id")
+    description = module.params.get("description")
+    subnets = module.params.get("subnets")
+    state = module.params.get("state")
+
+    # Remove duplicate subnets
+    if isinstance(subnets, list):
+        subnets = list(dict.fromkeys(subnets))
```

Comment:
Why the dict and list functions and not use set() function? Set return a itterable with unique values.

```suggestion
        subnets = set(subnets)
```

Should we provide a warning message that duplicates have been removed when there are duplicate provided?

Reply:
The dict function retains order while set does not. I preferred to keep subnets in a consistent order as it makes the tests a bit more predictable.

Reply:
Ok